django/core/exceptions.py

```diff
@@ -182,6 +184,30 @@ def __str__(self):
     def __repr__(self):
         return 'ValidationError(%s)' % self
 
+    def __eq__(self, other):
+        if not isinstance(other, ValidationError):
+            return NotImplemented
+        if hasattr(self, 'message'):
+            return (
+                hasattr(other, 'message') and
+                self.message == other.message and
+                self.code == other.code and
+                self.params == other.params
+            )   
+        return (
+            hasattr(self, 'error_list') == hasattr(other, 'error_list') and
+            hasattr(self, 'error_dict') == hasattr(other, 'error_dict') and
+            sorted(self.messages) == sorted(other.messages)
+        )
```

Comment:
I noticed that we missed keys comparison, e.g. `exception_a` and `exception_b` are equal:
```python
exception_a = ValidationError({'field1': 'field error', 'field2': 'err'})
exception_b = ValidationError({'field2': 'err', 'field1': 'field error'})
```
we should fix this, maybe:
```suggestion
        if hasattr(self, 'error_dict'):
            return hasattr(other, 'error_dict') and self.error_dict == other.error_dict
        return sorted(list(self)) == sorted(list(other))
```
a test is also needed.